lib/_http_agent.js

```diff
@@ -169,48 +172,59 @@ Agent.prototype.createSocket = function(req, options) {
 
   debug('createConnection', name, options);
   options.encoding = null;
-  var s = self.createConnection(options);
-  if (!self.sockets[name]) {
-    self.sockets[name] = [];
-  }
-  this.sockets[name].push(s);
-  debug('sockets', name, this.sockets[name].length);
+  options.oncreate = oncreate;
+  var called = false;
+  var newSocket = self.createConnection(options);
+  if (newSocket)
+    oncreate(null, newSocket);
```

Comment:
You mean put a `process.nextTick()` *inside* `oncreate()` that wraps the majority of that code? Otherwise `oncreate()` *could* be executed immediately within `createConnection()` too....